lib/registrations/registration_checker.rb

```diff
@@ -182,12 +182,21 @@ def contains_organizer_fields?(request, organizer_fields)
       def validate_update_status!(new_status, competition, current_user, target_user, registration, events)
         raise WcaExceptions::RegistrationError.new(:unprocessable_entity, Registrations::ErrorCodes::INVALID_REQUEST_DATA) unless
           Registration.competing_statuses.include?(new_status)
-        raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED) if
-          new_status == Registrations::Helper::STATUS_ACCEPTED && competition.competitor_limit_enabled? &&
-          competition.registrations.accepted.count >= competition.competitor_limit
         raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::ALREADY_REGISTERED_IN_SERIES) if
           new_status == Registrations::Helper::STATUS_ACCEPTED && existing_registration_in_series?(competition, target_user)
 
+        if new_status == Registrations::Helper::STATUS_ACCEPTED && competition.competitor_limit_enabled?
+          raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED) if
+            competition.registrations.competing_status_accepted.count >= competition.competitor_limit
+
+          if competition.enforce_newcomer_month_reservations? && !target_user.newcomer_month_eligible?
+            available_spots = competition.competitor_limit - competition.registrations.competing_status_accepted.count
```

Comment:
I think you can reuse the `competition.registrations.competing_status_accepted.count` part. If you wanna be very sneaky, you can even reuse the whole `available_spots` variable, and for the "general" check above you throw if `available_spots <= 0`.

Reply:
(but the latter is really just personal taste, feel free to keep it the way it is as you see fit.)